src/session.rs

```diff
@@ -111,9 +120,21 @@ impl<FS: Filesystem> Session<FS> {
         })
     }
 
-    /// Return path of the mounted filesystem
-    pub fn mountpoint(&self) -> &Path {
-        &self.mountpoint
+    /// Wrap an existing /dev/fuse file descriptor. This doesn't mount the
+    /// filesystem anywhere; that must be done separately.
+    pub fn from_fd(filesystem: FS, fd: OwnedFd, acl: SessionACL) -> Self {
```

Comment:
Instead of `SessionACL` I think this should take `&[MountOption]` and in the documentation state that it should be the same set of options that was used to create the mount and FD.

Can you also add to the documentation that the caller is responsible for unmounting the mountpoint?

Reply:
> Instead of `SessionACL` I think this should take `&[MountOption]` and in the documentation state that it should be the same set of options that was used to create the mount and FD.

Hm, I don't agree. A mount option like `fsname` has no effect here and will be ignored.

What if I added `SessionACL::from_mount_options(options: impl AsRef<[MountOption]>) -> Self`?

Reply:
Ok ya, I take it back. I'd been thinking there might be other `MountOption`s that are useful, but looking through the code again I see that they're all consumed when the mount is created.